README.md
# @36node/bus-chart-sdk

[![version][0]][1] [![downloads][2]][3]

主要负责历史数据的统计与存储。

## Install

```bash
yarn add @36node/bus-chart-sdk
```

## Usage

```js
const SDK = require("@36node/bus-chart-sdk");

const sdk = new SDK({ base: "http://localhost:3000/chart/v0", token: "secret" });
sdk.vehicle.listMileages({
      vehicleId: "xxxx",
      query: {
        filter: {
          at: {
            $gt: "2019-01-01",
            $lt: "2020-01-01",
          },
        },
      },
    });
```

For more usages, see `index.test.js`.

## Contributing

1. Fork it!
2. Create your feature branch: `git checkout -b my-new-feature`
3. Commit your changes: `git commit -am 'Add some feature'`
4. Push to the branch: `git push origin my-new-feature`
5. Submit a pull request :D

## Author

**@36node/bus-chart-sdk** © [36node](https://github.com/36node), Released under the [MIT](./LICENSE) License.

Authored and maintained by 36node with help from contributors ([list](https://github.com/36node/bus-chart-sdk-js/contributors)).

> GitHub [@36node](https://github.com/36node)

[0]: https://img.shields.io/npm/v/@36node/bus-chart-sdk.svg?style=flat
[1]: https://npmjs.com/package/@36node/bus-chart-sdk
[2]: https://img.shields.io/npm/dm/@36node/bus-chart-sdk.svg?style=flat
[3]: https://npmjs.com/package/@36node/bus-chart-sdk


CHANGELOG.md
# Changelog

All notable changes to this project will be documented in this file. See [standard-version](https://github.com/conventional-changelog/standard-version) for commit guidelines.

### [0.13.24](https://github.com/36node/bus-chart-sdk-js/compare/v0.13.23...v0.13.24) (2020-05-11)

### [0.13.23](https://github.com/36node/bus-chart-sdk-js/compare/v0.13.22...v0.13.23) (2020-05-06)

### [0.13.22](https://github.com/36node/bus-chart-sdk-js/compare/v0.13.21...v0.13.22) (2020-04-09)


### Features

* driver power agg api ([4fc9071](https://github.com/36node/bus-chart-sdk-js/commit/4fc9071))

### [0.13.21](https://github.com/36node/bus-chart-sdk-js/compare/v0.13.20...v0.13.21) (2020-04-08)

### [0.13.20](https://github.com/36node/bus-chart-sdk-js/compare/v0.13.19...v0.13.20) (2020-03-12)

### [0.13.19](https://github.com/36node/bus-chart-sdk-js/compare/v0.13.18...v0.13.19) (2020-02-12)

### [0.13.18](https://github.com/36node/bus-chart-sdk-js/compare/v0.13.17...v0.13.18) (2020-01-31)


### Features

* update auth ([57b3c28](https://github.com/36node/bus-chart-sdk-js/commit/57b3c28))

### [0.13.17](https://github.com/36node/bus-chart-sdk-js/compare/v0.13.16...v0.13.17) (2020-01-20)

### [0.13.16](https://github.com/36node/bus-chart-sdk-js/compare/v0.13.15...v0.13.16) (2020-01-19)


### Bug Fixes

* screen api params ([818b033](https://github.com/36node/bus-chart-sdk-js/commit/818b033))

### [0.13.15](https://github.com/36node/bus-chart-sdk-js/compare/v0.13.14...v0.13.15) (2020-01-19)


### Features

* upgrade openapi ([3e87773](https://github.com/36node/bus-chart-sdk-js/commit/3e87773))

### [0.13.14](https://github.com/36node/bus-chart-sdk-js/compare/v0.13.13...v0.13.14) (2020-01-19)

### [0.13.13](https://github.com/36node/bus-chart-sdk-js/compare/v0.13.12...v0.13.13) (2020-01-15)

### [0.13.12](https://github.com/36node/bus-chart-sdk-js/compare/v0.13.11...v0.13.12) (2020-01-15)

### [0.13.11](https://github.com/36node/bus-chart-sdk-js/compare/v0.13.10...v0.13.11) (2020-01-02)

### [0.13.10](https://github.com/36node/bus-chart-sdk-js/compare/v0.13.9...v0.13.10) (2019-12-30)

### [0.13.9](https://github.com/36node/bus-chart-sdk-js/compare/v0.13.8...v0.13.9) (2019-12-30)

### [0.13.8](https://github.com/36node/bus-chart-sdk-js/compare/v0.13.7...v0.13.8) (2019-12-27)

### [0.13.7](https://github.com/36node/bus-chart-sdk-js/compare/v0.13.6...v0.13.7) (2019-12-17)

### [0.13.6](https://github.com/36node/bus-chart-sdk-js/compare/v0.13.5...v0.13.6) (2019-12-06)


### Features

* update ([38619e3](https://github.com/36node/bus-chart-sdk-js/commit/38619e3))

### [0.13.5](https://github.com/36node/bus-chart-sdk-js/compare/v0.13.4...v0.13.5) (2019-11-14)

### [0.13.4](https://github.com/36node/bus-chart-sdk-js/compare/v0.13.3...v0.13.4) (2019-10-16)

### [0.13.3](https://github.com/36node/bus-chart-sdk-js/compare/v0.13.2...v0.13.3) (2019-10-14)


### Bug Fixes

* expo mjs extension err ([60ccc98](https://github.com/36node/bus-chart-sdk-js/commit/60ccc98))

### [0.13.2](https://github.com/36node/bus-chart-sdk-js/compare/v0.13.1...v0.13.2) (2019-09-10)

### [0.13.1](https://github.com/36node/bus-chart-sdk-js/compare/v0.13.0...v0.13.1) (2019-09-10)

## [0.13.0](https://github.com/36node/bus-chart-sdk-js/compare/v0.12.1...v0.13.0) (2019-09-10)


### Bug Fixes

* update discharge api ([c12a905](https://github.com/36node/bus-chart-sdk-js/commit/c12a905))


### Features

* add description ([11ceca1](https://github.com/36node/bus-chart-sdk-js/commit/11ceca1))



### [0.12.1](https://github.com/36node/bus-chart-sdk-js/compare/v0.12.0...v0.12.1) (2019-07-21)


### Bug Fixes

* rename mock alerts and vehicle ([680ff5f](https://github.com/36node/bus-chart-sdk-js/commit/680ff5f))



## [0.12.0](https://github.com/36node/bus-chart-sdk-js/compare/v0.11.0...v0.12.0) (2019-07-10)


### Features

* add chart alerts api ([c64b656](https://github.com/36node/bus-chart-sdk-js/commit/c64b656))



## [0.11.0](https://github.com/36node/bus-chart-sdk-js/compare/v0.10.1...v0.11.0) (2019-07-08)


### Bug Fixes

* test errors ([df0781d](https://github.com/36node/bus-chart-sdk-js/commit/df0781d))


### Features

* add /chart/vehicles api ([221b5e0](https://github.com/36node/bus-chart-sdk-js/commit/221b5e0))
* add /chart/vehicles api ([6375748](https://github.com/36node/bus-chart-sdk-js/commit/6375748))



### [0.10.1](https://github.com/36node/bus-chart-sdk-js/compare/v0.10.0...v0.10.1) (2019-07-07)



## [0.10.0](https://github.com/36node/bus-chart-sdk-js/compare/v0.9.0...v0.10.0) (2019-06-23)


### Features

* add report insulation warning data ([20bffad](https://github.com/36node/bus-chart-sdk-js/commit/20bffad))
* add report tire warning data ([8e77a31](https://github.com/36node/bus-chart-sdk-js/commit/8e77a31))



## [0.9.0](https://github.com/36node/bus-chart-sdk-js/compare/v0.8.1...v0.9.0) (2019-06-23)


### Features

* add battery warning report data ([40ad6ef](https://github.com/36node/bus-chart-sdk-js/commit/40ad6ef))



### [0.8.1](https://github.com/36node/bus-chart-sdk-js/compare/v0.8.0...v0.8.1) (2019-06-23)



## [0.8.0](https://github.com/36node/bus-chart-sdk-js/compare/v0.6.2...v0.8.0) (2019-06-22)


### Features

* update warnings report data structure ([7ec0272](https://github.com/36node/bus-chart-sdk-js/commit/7ec0272))



## 0.7.0 (2019-06-21)


### Features

* add mock data ([9a52c78](https://github.com/36node/bus-chart-sdk-js/commit/9a52c78))



## [0.7.0](https://github.com/36node/bus-chart-sdk-js/compare/v0.6.2...v0.7.0) (2019-06-21)


### Features

* add mock data ([9a52c78](https://github.com/36node/bus-chart-sdk-js/commit/9a52c78))



### [0.6.2](https://github.com/36node/bus-chart-sdk-js/compare/v0.6.1...v0.6.2) (2019-06-21)



### [0.6.1](https://github.com/36node/bus-chart-sdk-js/compare/v0.6.0...v0.6.1) (2019-06-20)



## [0.6.0](https://github.com/36node/bus-chart-sdk-js/compare/v0.5.0...v0.6.0) (2019-06-19)


### Features


src/index.test.js
import SDK from "./index";

const sdk = new SDK({
  base: "http://localhost:3000/chart/v0",
  token: "",
});

describe("## SDK vehicle mileages", () => {
  it("must provide vehicleId", () => {
    const listMileages = () => sdk.vehicle.listMileages();

    expect(listMileages).toThrow("vehicleId is required for listMileages");
  });

  it("should list records", async () => {
    const result = await sdk.vehicle.listMileages({
      vehicleId: "xxxx",
      query: {
        filter: {
          at: {
            $gt: "2019-01-01",
            $lt: "2020-01-01",
          },
        },
      },
    });

    expect(result.body.length).toBeGreaterThan(0);
    const mileageRecord = result.body[0];
    expect(mileageRecord.at).toBeTruthy();
    expect(mileageRecord.mileage).toBeGreaterThan(0);
  });
});


src/vehicle.test.js
import SDK from "./index";

const sdk = new SDK({
  base: "http://localhost:3000/chart/v0",
  token: "",
});

describe("## SDK vehicle mileages", () => {
  it("must provide vehicleId", () => {
    const listMileages = () => sdk.vehicle.listMileages();

    expect(listMileages).toThrow("vehicleId is required for listMileages");
  });

  it("should list records", async () => {
    const result = await sdk.vehicle.listMileages({
      vehicleId: "xxxx",
      query: {
        filter: {
          at: {
            $gt: "2019-01-01",
            $lt: "2020-01-01",
          },
        },
      },
    });

    expect(result.body.length).toBeGreaterThan(0);
    const mileageRecord = result.body[0];
    expect(mileageRecord.at).toBeTruthy();
    expect(mileageRecord.mileage).toBeGreaterThan(0);
  });
});

describe("## SDK vehicle energy consumptions", () => {
  it("must provide vehicleId", () => {
    const listEnergyConsumptions = () => sdk.vehicle.listEnergyConsumptions();

    expect(listEnergyConsumptions).toThrow(
      "vehicleId is required for listEnergyConsumptions"
    );
  });

  it("should list records", async () => {
    const result = await sdk.vehicle.listEnergyConsumptions({
      vehicleId: "xxxx",
      query: {
        filter: {
          at: {
            $gt: "2019-01-01",
            $lt: "2020-01-01",
          },
        },
      },
    });

    expect(result.body.length).toBeGreaterThan(0);
    const energyConsumptionRecord = result.body[0];
    expect(energyConsumptionRecord.at).toBeTruthy();
    expect(energyConsumptionRecord.discharge).toBeGreaterThan(0);
  });
});


src/warnings.test.js
import SDK from "./index";

const sdk = new SDK({
  base: "http://localhost:3000/chart/v0",
  token: "",
});

describe("## SDK warnings statistics", () => {
  it("must provide type", () => {
    const listMileages = () => sdk.warning.listWarningsStatistics();

    expect(listMileages).toThrow("type is required for listWarningsStatistics");
  });

  it("must provide groupKey", () => {
    const listMileages = () =>
      sdk.warning.listWarningsStatistics({ type: "alert" });

    expect(listMileages).toThrow(
      "groupKey is required for listWarningsStatistics"
    );
  });

  xit("should list records", async () => {
    const result = await sdk.warning.listWarningsStatistics({
      type: "alert",
      groupKey: "producer",
      query: {
        filter: {
          at: {
            $gt: "2019-01-01",
            $lt: "2020-01-01",
          },
        },
      },
    });

    expect(result.body.length).toBeGreaterThan(0);
    const record = result.body[0];
    expect(record.group).toBeTruthy();
    expect(record.name).toBeTruthy();
    expect(record.data).toBeTruthy();
  });
});


mock/alerts.js
/**
 * Mock alerts
 */

const faker = require("faker");
const moment = require("moment");
const _ = require("lodash");
const { Faults } = require("./constants");

const genChartAlerts = ({ vehicles = [], count }) => {
  const ret = [];

  // 生成两个月的数据
  const start = moment()
    .subtract(1, "month")
    .startOf("month");
  const end = moment()
    .add(1, "day")
    .startOf("day");

  let remainCount = count;
  for (let day = moment(end); day.isAfter(start); day.subtract(1, "day")) {
    const curCount = faker.random.number({ min: 3, max: 80 });

    _.range(curCount).forEach(index => {
      const vehicle = faker.random.arrayElement(vehicles);
      const lastAt = faker.date.between(
        moment(day)
          .startOf("day")
          .toDate(),
        moment(day)
          .endOf("day")
          .toDate()
      );
      const fault = faker.random.arrayElement(Faults.filter(f => f.level));
      ret.push({
        startedAt: faker.date
          .between(
            moment(day)
              .startOf("day")
              .toDate(),
            lastAt
          )
          .toISOString(),
        lastAt: lastAt.toISOString(),
        vehicle: vehicle.id,
        ns: vehicle.ns,
        line: vehicle.line,
        name: fault.name,
        level: fault.level,
        code: `${fault.code}`,
        plate: vehicle.plate,
        vehicleModel: vehicle.model,
        vehicleModelBrief: vehicle.modelBrief,
        vehicleNo: vehicle.no,
        vehicleMileage: faker.random.number({ min: 10000, max: 30000 }),
        vehicleYearsFromPlate: vehicle.lifeYear,
        state: faker.random.arrayElement(["OPEN", "CLOSE"]),
        count: faker.random.number({ min: 1, max: 30 }),
      });
    });

    remainCount -= curCount;
    if (remainCount < 0) {
      break;
    }
  }

  return ret;
};

module.exports = genChartAlerts;


mock/vehicle.js
const faker = require("faker");
const moment = require("moment");
const _ = require("lodash");

const listMileages = (vehicleId, { at_gt, at_lt }) => {
  const result = _.range(-500, 0)
    .map(i =>
      moment()
        .startOf("day")
        .add(i, "days")
    )
    .filter(at => at.diff(at_gt) >= 0 && at.diff(at_lt) <= 0)
    .map(at => ({
      at: at.format("YYYY-MM-DD"),
      mileage: faker.random.number({
        min: 0,
        max: 400,
        precision: 0.01,
      }),
    }));

  return result;
};

const listEnergyConsumptions = (vehicleId, { at_gt, at_lt }) => {
  const result = _.range(-500, 0)
    .map(i =>
      moment()
        .startOf("day")
        .add(i, "days")
    )
    .filter(at => at.diff(at_gt) >= 0 && at.diff(at_lt) <= 0)
    .map(at => ({
      at: at.format("YYYY-MM-DD"),
      discharge: faker.random.number({
        min: 0,
        max: 400,
        precision: 0.01,
      }),
    }));

  return result;
};

const genChartVehicles = (vehicles = []) => {
  // 每辆车生成今年的数据, 每日一条数据

  const ret = [];
  const start = moment().startOf("year");
  const end = moment()
    .add(1, "day")
    .startOf("day");

  for (const vehicle of vehicles) {
    // 按每天循环
    let lastMilesage = faker.random.number({ min: 10000, max: 30000 });
    let lastConsumption =
      (lastMilesage / 100) * faker.random.number({ min: 40, max: 60 });

    for (let day = moment(start); day.isBefore(end); day.add(1, "day")) {
      const mileageIncrement = faker.random.number({ min: 50, max: 200 });
      const consumptionIncrement = faker.random.number({ min: 40, max: 60 });

      lastMilesage += mileageIncrement;
      lastConsumption += consumptionIncrement;

      ret.push({
        at: day.toISOString(),
        vin: vehicle.id,
        ns: vehicle.ns,
        line: vehicle.line,
        no: vehicle.no,
        producer: vehicle.producer,
        modelBrief: vehicle.modelBrief,
        model: vehicle.model,
        mileage: lastMilesage,
        consumption: lastConsumption,
      });
    }
  }

  return ret;
};

const defaultVehicles = [
  {
    id: "FAKEMFVQFQYPT1O5C",
    vin: "FAKEMFVQFQYPT1O5C",
    brands: "LLC",
    capacity: 57,
    emission: "UNKNOWN",
    engineNo: "FAKE0AHTUZVP",
    iccid: "FAKE1Z31RY9A",
    length: 9,
    lifeYear: 12,
    line: "119路",
    model: "SLK6109ULE0BEVX",
    modelBrief: "L9E",
    modified: true,
    no: "L9E-CMUV",
    place: "42265 Ferne Point",
    plate: "沪CFAKEYCXLV",
    plateAt: "2019-01-21T09:21:56.193Z",
    powerBy: "混合动力",
    producer: "申龙",
    purchaseAt: "2018-09-15T11:20:38.898Z",
    remark: "expedita delectus quia",
    scrapped: false,
    seats: 48,
    type: "大型普通客车",
    validTill: "2031-01-21T09:21:56.193Z",
    ns: "/bus/company1",
    createdAt: "2019-03-02T04:01:17.391Z",
    updatedAt: "2019-10-27T14:18:26.536Z",
    state: "ONLINE",
  },
  {
    id: "FAKE1ILZF19KPY6N2",
    vin: "FAKE1ILZF19KPY6N2",
    brands: "Group",
    capacity: 36,
    emission: "UNKNOWN",
    engineNo: "FAKENFJM3V1C",
    iccid: "FAKE4383T01Q",
    length: 14,
    lifeYear: 11,
    line: "浦东62路",
    model: "SLK6109ULE0BEVX",
    modelBrief: "L0B",
    modified: true,
    no: "L0B-W9VY",
    place: "163 Tod Crossroad",
    plate: "沪AFAKE4CEQ6",
    plateAt: "2019-06-05T00:54:37.057Z",
    powerBy: "纯电动",
    producer: "申龙",
    purchaseAt: "2019-04-03T16:28:42.261Z",
    remark: "repellat magni similique",
    scrapped: true,
    seats: 52,
    type: "大型普通客车",
    validTill: "2030-06-05T00:54:37.057Z",
    ns: "/bus",
    createdAt: "2018-11-16T22:31:47.233Z",
    updatedAt: "2020-01-05T05:40:16.774Z",
    state: "ONLINE",
  },
  {
    id: "FAKERIMVG7YH35HKG",
    vin: "FAKERIMVG7YH35HKG",
    brands: "Group",
    capacity: 44,
    emission: "UNKNOWN",
    engineNo: "FAKE8PTY4VQ6",
    iccid: "FAKE2G95RJI0",
    length: 8,
    lifeYear: 11,
    line: "浦东1路",
    model: "ZK6105BEVG29",
    modelBrief: "SY",
    modified: false,
    no: "SY-PZKD",
    place: "2172 Reed Landing",
    plate: "沪AFAKE446KJ",
    plateAt: "2018-12-03T21:28:13.595Z",
    powerBy: "混合动力",
    producer: "宇通",
    purchaseAt: "2018-09-19T12:09:17.382Z",
    remark: "non doloremque autem",
    scrapped: true,
    seats: 58,
    type: "大型普通客车",
    validTill: "2029-12-03T21:28:13.595Z",
    ns: "/bus/company1/sub1",
    createdAt: "2019-03-27T23:33:04.212Z",
    updatedAt: "2020-05-27T20:41:30.163Z",
    state: "UNKNOWN",
  },
  {
    id: "FAKE6AIOJOSPQB85K",
    vin: "FAKE6AIOJOSPQB85K",
    brands: "LLC",
    capacity: 39,
    emission: "UNKNOWN",
    engineNo: "FAKEC66I5XT9",
    iccid: "FAKE6OKCUAP4",
    length: 15,
    lifeYear: 10,
    line: "611路",
    model: "ZK6125BEVG6",
    modelBrief: "Z0A",
    modified: true,
    no: "Z0A-AAFV",
    place: "480 Isaac Walk",
    plate: "沪AFAKEWUS7E",
    plateAt: "2019-02-25T18:19:46.285Z",
    powerBy: "混合动力",
    producer: "宇通",
    purchaseAt: "2019-06-15T11:09:56.942Z",
    remark: "omnis repellat et",
    scrapped: false,
    seats: 45,
    type: "大型普通客车",
    validTill: "2029-02-25T18:19:46.285Z",
    ns: "/bus/company2",
    createdAt: "2019-07-03T19:16:02.104Z",
    updatedAt: "2019-08-30T00:39:17.660Z",
    state: "GPS_ERROR",
  },
  {
    id: "FAKEQWEE5TI8D8TVY",
    vin: "FAKEQWEE5TI8D8TVY",
    brands: "Group",
    capacity: 48,
    emission: "UNKNOWN",
    engineNo: "FAKEROKEYGH2",
    iccid: "FAKEGPY1RXTA",
    length: 11,
    lifeYear: 9,
    line: "7路",
    model: "SXC6120GHEV",
    modelBrief: "W1C",
    modified: false,
    no: "W1C-I93H",
    place: "34966 Vandervort Mills",
    plate: "沪AFAKELCOTV",
    plateAt: "2018-10-06T20:41:58.769Z",
    powerBy: "混合动力",
    producer: "万象",
    purchaseAt: "2018-08-14T05:33:48.152Z",
    remark: "reprehenderit voluptatem et",
    scrapped: false,
    seats: 39,
    type: "大型普通客车",
    validTill: "2027-10-06T20:41:58.769Z",
    ns: "/bus/company1/sub2",
    createdAt: "2018-11-28T05:08:30.474Z",
    updatedAt: "2020-05-15T11:23:38.090Z",
    state: "GPS_ERROR",
  },
  {
    id: "FAKEP40H9YZ5GC236",
    vin: "FAKEP40H9YZ5GC236",
    brands: "and Sons",
    capacity: 47,
    emission: "UNKNOWN",
    engineNo: "FAKEKZ5GXZOB",
    iccid: "FAKEFPRCIJZ4",
    length: 7,
    lifeYear: 9,
    line: "浦东1路",
    model: "ZK6125BEVG25",
    modelBrief: "Z2E",
    modified: false,
    no: "Z2E-BWEE",
    place: "5932 Blaise Path",
    plate: "沪BFAKEBTTDB",
    plateAt: "2018-08-17T18:39:05.069Z",
    powerBy: "纯电动",
    producer: "宇通",
    purchaseAt: "2019-04-06T07:37:02.419Z",
    remark: "omnis cum saepe",
    scrapped: false,
    seats: 48,
    type: "大型普通客车",
    validTill: "2027-08-17T18:39:05.069Z",
    ns: "/bus/company1/sub1",
    createdAt: "2018-12-24T03:01:28.370Z",
    updatedAt: "2020-04-10T17:53:41.255Z",
    state: "OFFLINE",
  },
  {
    id: "FAKEMW2U2ELDN1ZWK",
    vin: "FAKEMW2U2ELDN1ZWK",
    brands: "LLC",
    capacity: 33,
    emission: "UNKNOWN",
    engineNo: "FAKE75XRIXMZ",
    iccid: "FAKEURYQ0Y6X",
    length: 13,
    lifeYear: 10,
    line: "52路(外环路)",
    model: "SWB6128EV56",
    modelBrief: "S2U",
    modified: false,
    no: "S2U-09W0",
    place: "18874 Corrine Centers",
    plate: "沪AFAKEG4WHO",
    plateAt: "2019-02-24T14:45:56.632Z",
    powerBy: "纯电动",
    producer: "申沃",
    purchaseAt: "2018-10-19T05:40:37.419Z",
    remark: "sapiente reiciendis magnam",
    scrapped: false,
    seats: 58,
    type: "大型普通客车",
    validTill: "2029-02-24T14:45:56.632Z",
    ns: "/bus/company1/sub1",
    createdAt: "2019-03-27T10:59:49.009Z",
    updatedAt: "2020-06-28T00:41:46.213Z",
    state: "OFFLINE",
  },
  {
    id: "FAKEEQDQ658ZX3U10",
    vin: "FAKEEQDQ658ZX3U10",
    brands: "Inc",
    capacity: 43,
    emission: "UNKNOWN",
    engineNo: "FAKEMJITBBMM",
    iccid: "FAKEE4BWVKTO",
    length: 9,
    lifeYear: 12,
    line: "浦东2路",
    model: "SLK6929ULE0BEVS1",
    modelBrief: "L0B",
    modified: true,
    no: "L0B-S2CN",
    place: "617 Wyatt Ramp",
    plate: "沪DFAKE78Z6Z",
    plateAt: "2019-05-22T19:29:41.654Z",
    powerBy: "纯电动",
    producer: "申龙",
    purchaseAt: "2018-11-07T23:35:25.304Z",
    remark: "quae fugiat vel",
    scrapped: false,
    seats: 47,
    type: "大型普通客车",
    validTill: "2031-05-22T19:29:41.654Z",
    ns: "/bus",
    createdAt: "2018-08-05T03:24:55.742Z",
    updatedAt: "2020-02-19T00:54:09.234Z",
    state: "OFFLINE",
  },
  {
    id: "FAKETDPYMMW6FWC84",
    vin: "FAKETDPYMMW6FWC84",
    brands: "Group",
    capacity: 42,
    emission: "UNKNOWN",
    engineNo: "FAKEFWG16DXD",
    iccid: "FAKEJH8YENV4",
    length: 9,
    lifeYear: 10,
    line: "62路",
    model: "SXC6105GBEV",
    modelBrief: "W1D",
    modified: true,
    no: "W1D-DETD",
    place: "580 Selena Tunnel",
    plate: "沪CFAKEIO0O3",
    plateAt: "2018-09-25T03:41:39.482Z",
    powerBy: "纯电动",
    producer: "万象",
    purchaseAt: "2019-07-03T07:11:45.777Z",
    remark: "autem non hic",
    scrapped: true,
    seats: 44,
    type: "大型普通客车",
    validTill: "2028-09-25T03:41:39.482Z",
    ns: "/bus/company1/sub2",
    createdAt: "2018-10-10T11:37:27.805Z",
    updatedAt: "2020-03-29T05:57:57.302Z",
    state: "ONLINE",
  },
  {
    id: "FAKEZWUXJ0O77KPLE",
    vin: "FAKEZWUXJ0O77KPLE",
    brands: "and Sons",
    capacity: 61,
    emission: "UNKNOWN",
    engineNo: "FAKEPAU41F0V",
    iccid: "FAKE5FZOXC4W",
    length: 10,
    lifeYear: 11,
    line: "浦东1路",
    model: "SXC6110GBEV3",
    modelBrief: "W1C",
    modified: true,
    no: "W1C-44YT",
    place: "834 King Light",
    plate: "沪AFAKE7KJAN",
    plateAt: "2019-01-27T22:28:43.360Z",
    powerBy: "纯电动",
    producer: "万象",
    purchaseAt: "2018-08-25T02:41:13.489Z",
    remark: "voluptatem iusto et",
    scrapped: false,
    seats: 58,
    type: "大型普通客车",
    validTill: "2030-01-27T22:28:43.360Z",
    ns: "/bus/company1/sub1",
    createdAt: "2018-10-11T08:01:11.035Z",
    updatedAt: "2019-09-02T01:53:05.774Z",
    state: "GPS_ERROR",
  },
];

module.exports = {
  listMileages,
  listEnergyConsumptions,
  genChartVehicles,
  defaultVehicles,
};
